pkg/virt-handler/rest/lifecycle.go

```diff
@@ -147,6 +147,23 @@ func (lh *LifecycleHandler) UnfreezeHandler(request *restful.Request, response *
 	response.WriteHeader(http.StatusAccepted)
 }
 
+func (lh *LifecycleHandler) ResetHandler(request *restful.Request, response *restful.Response) {
+	vmi, client, err := lh.getVMILauncherClient(request, response)
+	if err != nil {
+		return
+	}
+
+	err = client.ResetVirtualMachine(vmi)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("Failed to reset VMI")
+		response.WriteError(http.StatusInternalServerError, err)
```

Comment:
Some Handlers fail with `StatusBadRequest`, some with `StatusInternalServerError`. What is more applicable?

Reply:
`StatusInternalServerError` is consistent with the soft reboot behavior.

Reply:
ACK, it was more of a general question though.

Reply:
>ACK, it was more of a general question though.

 StatusInternalServerError is more applicable for this response. bad request indicates there is an issue with the client side making the request, which doesn't appear to be the case here.